src/HoldRequestResultConsumerListener.php

```diff
@@ -244,10 +244,15 @@ protected function processListenerEvents()
                         }
                     }
                 }
+
             } catch (\Exception $exception) {
                 APILogger::addError(
-                    'Exception thrown: ' . $exception->getMessage()
+                    'Exception thrown: ' . $exception->getMessage() .
+                    ', Error code: ' . $exception->getCode()
                 );
+                if ($exception->getCode() >= 500 && $exception->getCode() <= 599) {
```

Comment:
Instead of checking the code returned here, you an add a catch for `RetriableAPIException`.